conan/tools/gnu/autotools.py

```diff
@@ -13,7 +14,7 @@ def __init__(self, conanfile):
         self._configure_args = toolchain_file_content.get("configure_args")
         self._make_args = toolchain_file_content.get("make_args")
 
-    def configure(self):
+    def configure(self, source_folder=None):
```

Comment:
The current ``tools.cmake.CMake`` implementation is probably a legacy one, with the ``source_folder``.
I understand the use case, maybe it is just the name that is confusing. The ``source_folder`` is already defined by the layout. Maybe we should rename it to something like ``configure(build_script_folder=xxxx)``? (Just an idea, could be ``configure_folder``, ``cmakelists_folder``, or any other thing).

Reply:
totally agree